src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java

```diff
@@ -0,0 +1,308 @@
+            Object targetKey = fieldMapEntry.getValue();
+            if (targetKey instanceof Map) {
+                // call this method recursively when target key is a map
+                Object sourceObject = sourceAndMetadataMap.get(originalKey);
+                if (sourceObject instanceof List) {
+                    List<Object> sourceObjectList = (List<Object>) sourceObject;
+                    for (Object source : sourceObjectList) {
+                        if (source instanceof Map) {
+                            chunkCount = chunkMapType(
+                                (Map<String, Object>) source,
+                                (Map<String, Object>) targetKey,
+                                runtimeParameters,
+                                chunkCount
+                            );
+                        }
+                    }
+                } else if (sourceObject instanceof Map) {
+                    chunkCount = chunkMapType(
+                        (Map<String, Object>) sourceObject,
+                        (Map<String, Object>) targetKey,
+                        runtimeParameters,
+                        chunkCount
+                    );
+                }
+            } else {
+                // chunk the object when target key is a string
+                Object chunkObject = sourceAndMetadataMap.get(originalKey);
+                List<String> chunkedResult = new ArrayList<>();
+                chunkCount = chunkLeafType(chunkObject, chunkedResult, runtimeParameters, chunkCount);
+                sourceAndMetadataMap.put(String.valueOf(targetKey), chunkedResult);
+            }
+        }
+        return chunkCount;
+    }
+
+    private int chunkString(final String content, List<String> result, final Map<String, Object> runTimeParameters, int chunkCount) {
+        // chunk the content, return the updated chunkCount and add chunk passages into result
+        List<String> contentResult = chunker.chunk(content, runTimeParameters);
+        chunkCount += contentResult.size();
+        if (maxChunkLimit != DEFAULT_MAX_CHUNK_LIMIT && chunkCount > maxChunkLimit) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "Unable to chunk the document as the number of chunks [%s] exceeds the maximum chunk limit [%s]",
+                    chunkCount,
+                    maxChunkLimit
+                )
+            );
+        }
+        result.addAll(contentResult);
+        return chunkCount;
+    }
+
+    private int chunkList(
+        final List<String> contentList,
+        List<String> result,
+        final Map<String, Object> runTimeParameters,
+        int chunkCount
+    ) {
+        // flatten original output format from List<List<String>> to List<String>
+        for (String content : contentList) {
+            chunkCount = chunkString(content, result, runTimeParameters, chunkCount);
```

Comment:
same comment as above, can we have new int variable instead of reusing same one

Reply:
Will be changed.